docs/advanced/classes.rst

```diff
@@ -695,6 +695,13 @@ throwing a type error.
 
     To use the more convenient ``py::self`` notation, the additional
     header file :file:`pybind11/operators.h` must be included.
+    
+.. note::
+
+     A class that overrides ``__eq__()`` and does not define ``__hash__()`` will have its ``__hash__()`` implicitly set. 
```

Comment:
Suggestion: When a class overrides ``__eq__()`` and does not define ``__hash__()``, pybind11 will automatically add a ``__hash__()``.  (I think this rewording makes it more clear that this behavior is something pybind11 does.)